tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/NegativeTypeTestExprTest.java

```diff
@@ -0,0 +1,747 @@
+                "incompatible types: 'xml:Text' will not be matched to 'string'",
+                272, 9);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'xml<xml<xml:Text>>' will not be matched to 'string'",
+                273, 9);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'xml<xml<never>>' will not be matched to 'string'",
+                274, 9);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'xml' will not be matched to 'string'",
+                275, 9);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: '(Baz|int)' will not be matched to 'Bar'", 280, 17);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: '(Baz|int)' will not be matched to 'Qux'", 281, 18);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'Bar' will not be matched to 'Baz'", 284, 18);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'Bar' will not be matched to 'Quux'", 285, 18);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'Qux' will not be matched to 'Baz'", 288, 18);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'Qux' will not be matched to 'Quux'", 289, 18);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'Quux' will not be matched to 'Bar'", 292, 18);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'Quux' will not be matched to 'Qux'", 293, 18);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'Quux' will not be matched to 'record {| int i; boolean b; |}'", 294, 18);
+        BAssertUtil.validateError(negativeResult, i++,
+                "incompatible types: 'ClosedRecordWithIntField' will not be matched to " +
+                        "'record {| int i; string s; |}'", 297, 19);
+        BAssertUtil.validateError(negativeResult, i++, "incompatible types: 'object { }[]' will not be matched to " +
+                "'anydata'", 330, 10);
+        BAssertUtil.validateError(negativeResult, i++, "incompatible types: 'anydata' will not be matched to 'object " +
+                "{ }[]'", 336, 10);
+        BAssertUtil.validateError(negativeResult, i++, "incompatible types: 'Record' will not be matched to " +
+                "'RecordWithIntFieldAndNeverRestField'", 358, 18);
+        BAssertUtil.validateError(negativeResult, i, "incompatible types: 'Record' will not be matched to " +
+                "'RecordWithIntFieldAndEffectivelyNeverRestField'", 359, 18);
+        Assert.assertEquals(negativeResult.getErrorCount(), 26);
+    }
+
+    @Test
+    public void testValueTypeInUnion() {
```

Comment:
Some of the tests can be replaced with a parameterized test, But not sure whether it's a good option here.

Reply:
Will check on that